gpt_bpe.go

```diff
@@ -898,11 +908,48 @@ func (encoder *GPTEncoder) makeWordSplitter(
 				}
 
 				runeAccumulator = append(runeAccumulator, r)
-				specialsNode, specialToken = specialsRuneRoot.evaluate(
-					specialsNode, r)
-				if specialToken || r == '\n' {
+
+				if r == '\n' {
+					break
+				}
+
```

Comment:
This code now considers multiple candidates rather than a single exclusive candidate.